subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -580,14 +581,14 @@ public void setArtifactHandler(ArtifactHandler artifactHandler) {
     public abstract RepositoryHandler getRepositories();
 
     @Override
-    public ConfigurationContainer getConfigurations() {
+    public RoleBasedConfigurationContainerInternal getConfigurations() {
         if (configurationContainer == null) {
-            configurationContainer = services.get(ConfigurationContainer.class);
+            configurationContainer = services.get(RoleBasedConfigurationContainerInternal.class);
         }
         return configurationContainer;
     }
 
-    public void setConfigurationContainer(ConfigurationContainer configurationContainer) {
+    public void setConfigurationContainer(RoleBasedConfigurationContainerInternal configurationContainer) {
```

Comment:
This method seems weird to have. It looks like its only used in `NewDefaultProjectTest`. Do you think there is a way to avoid it in the test and get rid of this method entirely?

Reply:
Interesting, the same seems to be true of `artifactHandler` and `dependencyHandler`.  Why expose setters on these at all, if the only thing we use them for is to test that if set the set value is used?

I'm going to throw up a quick separate PR to remove these.